tools/integration_tests/write_large_files/random_write_large_file_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package write_large_files
+
+import (
+	"crypto/rand"
+	"log"
+	rand2 "math/rand"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+const NumberOfRandomWriteCalls = 20
+const DirForRandomWrite = "dirForRandomWrite"
+const MinWritableByteFromFile = 0
+const MaxWritableByteFromFile = 500 * OneMB
+const FileDownloadedFromBucket = "fileDownloadedFromBucket"
+const FiveHundredMBFileForRandomWriteInLocalSystem = "fiveHundredMBFileForRandomWriteInLocalSystem"
+
+func TestWriteLargeFileRandomly(t *testing.T) {
+	randomWriteDir := path.Join(setup.MntDir(), DirForRandomWrite)
+	err := os.Mkdir(randomWriteDir, setup.FilePermission_0600)
+	if err != nil {
+		t.Errorf("Error in creating directory:%v", err)
+	}
+	filePath := path.Join(randomWriteDir, FiveHundredMBFile)
+
+	f, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|syscall.O_DIRECT, setup.FilePermission_0600)
+	if err != nil {
+		t.Errorf("Open file for write at start: %v", err)
+		return
+	}
+
+	for i := 0; i < NumberOfRandomWriteCalls; i++ {
+		offset := rand2.Int63n(MaxWritableByteFromFile - MinWritableByteFromFile)
+
+		// Generate chunk with random data.
+		chunk := make([]byte, ChunkSize)
+		_, err := rand.Read(chunk)
+		if err != nil {
+			log.Fatalf("error while generating random string: %s", err)
+		}
+
+		// Write data in the file.
+		n, err := f.WriteAt(chunk, offset)
```

Comment:
I am not sure if WriteAt can beyond the existing file-size. If it can't then this can fail unnecessarily if `MaxWritableByteFromFile + chunkSize` >= 500M.